python/test/test_serialization.py

```diff
@@ -104,6 +104,59 @@ def test_version_mismatch_warning():
 # in this case it needs to wrap the tests.
 pl = import_optional_dependency('polars', raise_error=False)
 if pl is not None:
+    # member() will warn if instance is not even of the carrier type,
+    # but that behavior is tested elsewhere, and can be ignored here.
+    @pytest.mark.filterwarnings("ignore::UserWarning")
+    @pytest.mark.parametrize(
+        "_readable_name,dp_domain,in_value,out_value",
+        [
+            (str(dp_domain), dp_domain, in_value, out_value)
```

Comment:
Instead of the comprehension, would it be better to change the repr?

Going by Python conventions, I thought it was the case that repr was more gritty and str gave a pretty summary. Are we following python conventions?

Reply:
Pytest is somehow stringifying the input in a way we don't expect, because we have `__repr__` implemented which should handle repr and str pretty stringification.

Reply:
Sorry: I misdiagnosed this. I'll file an issue just to check on the __repr__ (since it seems like something was out of sync when we looked at the code), but it has nothing to do with this.

Pytest by default will only render primitives like strings and numbers. For any other object, it will just sequentially number it to give the test a name... but we can override that with `ids`, which I've just done for all of these tests.